src/kbucket.rs

```diff
@@ -115,34 +119,90 @@ impl KbucketTable {
                 }
             }
         }
-        SearchResult {
-            found: false,
-            bucket_index,
-            column_index: last_empty_index,
-        }
+        Search::Failure(bucket_index, last_empty_index)
     }
 
-    fn find_bucket_index(&self, identifier: Identifier) -> usize {
+    fn xor_bucket_index(&self, identifier: Identifier) -> usize {
         let x = U256::from(self.local_node_id);
         let y = U256::from(identifier);
         let xor_distance = x ^ y;
 
-        let bucket_index = MAX_BUCKETS - (xor_distance.leading_zeros() as usize);
-        println!(
-            "Xor distance leading zeros, {}",
-            xor_distance.leading_zeros()
-        );
-        println!("Bucket index for given key: {}", bucket_index);
-        bucket_index
+        println!("Identifier: {:?}", identifier);
+        println!("Leading zeros: {}", xor_distance.leading_zeros());
+        println!("\n");
+        MAX_BUCKETS - ((xor_distance.leading_zeros() - 1) as usize)
     }
 }
 
-/// TODO:  Implement real deal tests!
-///
-/// Test find_node()      **Requires adding nodes to our table**
-/// Test search_table()
-/// Test add_node()
 #[cfg(test)]
 mod tests {
     use super::*;
+    use crate::helper::testing::{mk_nodes, mk_table};
+
+    // Run "cargo test add_node -- --nocapture" to see that node was added to routing table.
+    #[test]
+    fn add_node() {
+        let dummy_nodes = mk_nodes();
+        let mut table = mk_table(dummy_nodes.clone());
```

Comment:
no reason to close all these nodes; I'd change the API here

```suggestion
        let mut table = mk_table(&dummy_nodes);
```

Reply:
Deleting this guy anyway

Reply:
But check out my return values for mk_nodes( ):  https://github.com/EchoAlice/my_kademlia/commit/c7aaf37239813e69b2f63df3de8dfef117b7fdad#diff-1a035035906977e355bc8da311ae54e7161dd36d85dd39bf81cbae9dabbf339aR149

What would you suggest doing?